rosidl_generator_cpp/rosidl_generator_cpp/__init__.py

```diff
@@ -114,10 +114,14 @@ def msg_type_to_cpp(type_):
             (type_.pkg_name, type_.type)
 
     if type_.is_array:
-        if type_.array_size is None:
+        if not type_.array_size:
```

Comment:
An array size of zero is not valid. So the check is the same as checking for `None`.